adage/analyze/api.py

```diff
@@ -427,7 +427,14 @@ def apply_filters(self, request, applicable_filters):
                 raise BadRequest("Invalid gene IDs: %s" % genes)
             # "in" operator in Django supports both list and set.
             qset = Q(gene1__in=ids) | Q(gene2__in=ids)
-            object_list = object_list.filter(qset).distinct()
+            direct_edges = Edge.objects.filter(qset).distinct()
+            related_genes = set()
+            for e in direct_edges:
+                related_genes.add(e.gene1)
+                related_genes.add(e.gene2)
+            object_list = object_list.filter(
+                gene1__in=related_genes, gene2__in=related_genes
+            ).distinct()
```

Comment:
Upon Deb's request, the left slider of correlation is set at 0 for now. When issue #184 is fixed, it will be set at the cutoff value of the user-selected machine learning model.